### ISSUE TITLE
Thanks for this!

### ISSUE THREAD

**Issue by krackers** [Wowfunhappy/CodeRunner-Mavericks-Workarounds#1]
Thanks for this! I actually ran into this sometime back but never thought to swizzle `signatureWithObjCTypes` (then again, 2.3 was fine for me so I never bothered too much).

One thing you might want to do is to get autocomplete for go working, add the the `libMacportsLegacySupport`  into  the `gocode` binary (in `SharedSupport/Developer/bin/gocode`).

**Comment by Wowfunhappy** [comment]
You're very welcome! I was also using version 2, but needed to be able to drag and drop tabs!

A couple questions, while I have you:

Were you able to use 2.3 on Mavericks without issues? I had to use 2.2.2, in order to avoid two problems:
- CodeRunner 2.3 would sometimes crash while quitting. While this is certainly the least disruptive time for an app to crash, I'd rather it not happen at all, because it caused OS X's crash dialog to pop up.
- In CodeRunner 2.3, a translucent black rectangle would sometimes appear in front of part of the window.
I found out this morning that the second issue seems to still be present in CodeRunner 3.1, but since it's intermittent I don't know if I'll be able to track it down...

And, is adding `libMacportsLegacySupport.dylib` to the `gocode` binary with `insert_dylib` sufficient to make it run? In the past, I've had to use `DYLD_INSERT_LIBRARIES` in order to launch stuff compiled with Go, which is annoyingly not automatic.

**Comment by krackers** [comment]
Yes you're right I've experienced some occasional crashes on 2.3 when quitting (and there's also another bug I've ran into where "run" doesn't show any console output, but both of these are too minor and infrequent for me to care about them). I also never realized you couldn't drag/drop tabs in 2.3; I guess since I mainly use coderunner for one-off quick scripts this didn't affect my workflow.

As for Go binaries, all of the ones I've tried worked fine using `insert_dylib`. I'm surprised that it only works using `DYLD_INSERT_LIBRARIES` for you; because we're just providing a symbol that doesn't exist rather than overriding an existing one, you shouldn't need `DYLD_INSERT_LIBRARIES`

 (P.S. I first learned about the existence of the macports support library from your post on macrumors – huge thanks for that, since before that I'd been manually recompiling things with go 1.10 which got increasingly complicated as dependent libraries kept updating).
